src/main/java/com/laytonsmith/core/functions/Threading.java

```diff
@@ -586,4 +587,212 @@ public Version since() {
+					+ " When a thread is interrupted, its interrupt status is set to true."
+					+ " This status can be checked using the " + new x_is_interrupted().getName() + " and " + new x_clear_interrupt().getName() + " function."
+					+ " Note that some blocking functions will throw an InterruptedException when the thread on which they are executed is interrupted.";
+		}
+	}
+
+	@api
+	@seealso({x_interrupt.class, x_clear_interrupt.class})
+	public static class x_is_interrupted extends AbstractFunction {
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return null;
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			if(args.length == 1) {
+				Thread th = THREAD_ID_MAP.get(args[0].val());
+				if(th != null) {
+					return CBoolean.get(th.isInterrupted());
+				} else {
+					return CBoolean.FALSE;
+				}
+			} else {
+				return CBoolean.get(Thread.currentThread().isInterrupted());
+			}
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+		@Override
+		public String getName() {
+			return "x_is_interrupted";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{0, 1};
+		}
+
+		@Override
+		public String docs() {
+			return "boolean {[string id]} Tests whether this thread with the given id has been interrupted."
+					+ " If id not passed, it is be current thread."
+					+ " The \"interrupt status\" of the thread is unaffected by this method."
+					+ " In case if the thread does not exist or is not alive, return false";
+		}
+	}
+
+	@api
+	@seealso({x_interrupt.class, x_is_interrupted.class})
+	public static class x_clear_interrupt extends AbstractFunction {
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return null;
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			return CBoolean.get(Thread.interrupted());
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+		@Override
+		public String getName() {
+			return "x_clear_interrupt";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{0};
+		}
+
+		@Override
+		public String docs() {
+			return "boolean {} Tests whether this thread has been interrupted."
+					+ " The \"interrupt status\" of the thread is cleared by this method. In other words, if this method were to be called twice in succession, the second call would return false"
+					+ " (unless the current thread were interrupted again, after the first call had cleared its \"interrupt status\" and before the second call had examined it).";
```

Comment:
"Clears the interrupt status of the given thread, or the thread provided. If this call actually cleared the interrupt status, true is returned, if the thread was already not interrupted, false is returned."

Actually though, this brings up another point, why is this the only method that doesn't actually accept a thread id? Unless I'm missing some reason, it should.

Reply:
@LadyCailin Java only has a static method that clears the interrupt flag on the current thread if only API is considered. There is a non-synchronized private native method that can be called to clear the interrupt flag for a given thread, but I wouldn't feel comfortable depending on that. When you think about it, as soon as a thread is interrupted, that thread is specifically signalled and should be handling its own interrupt flag. I can't think of a sane usage for clearing the interrupt flag on another thread, everything I come up with fits `import()/export()` better.

Reply:
Hmm, fair enough. Then my provided documentation string should be slightly modified, but otherwise this seems good.